### PR TITLE
Return response trailers from test server

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses #6880

### CODE REVIEW

File: src/Hosting/TestHost/src/ResponseStream.cs [hunk 1]
```diff
@@ -116,6 +117,7 @@ public async override Task<int> ReadAsync(byte[] buffer, int offset, int count,
                 if (result.Buffer.IsEmpty && result.IsCompleted)
                 {
                     _pipe.Reader.Complete();
+                    _readComplete();
```
Comment (Tratcher) [inline src/Hosting/TestHost/src/ResponseStream.cs]: ReadAsync can be called multiple times at the end and return 0 each time, no? Or would the pipe reader throw if you did that? You don't want _readComplete being called multiple times.

File: src/Hosting/TestHost/test/ClientHandlerTests.cs [hunk 2]
```diff
@@ -88,6 +88,55 @@ public Task SingleSlashNotMovedToPathBase()
             return httpClient.GetAsync("https://example.com/");
         }
 
+        [Fact]
+        public async Task ServerTrailersSetOnResponseAfterContentRead()
+        {
+            var tcs = new TaskCompletionSource<object>();
```
Comment (Tratcher) [inline src/Hosting/TestHost/test/ClientHandlerTests.cs]: RunContinuationAsync

**Review by Tratcher — APPROVED** [review]
Nice and clean.

File: src/Hosting/TestHost/src/ResponseStream.cs (hunk 1 above)
Comment (JamesNK) [reply src/Hosting/TestHost/src/ResponseStream.cs]: The pipe reader throws. I've updated the stream to return 0 when the reader is complete.

Can you double check the way I've done it?

**Review by Tratcher — APPROVED** [review]

**Pull request merged**